js/coinex.js

```diff
@@ -452,6 +452,78 @@ module.exports = class coinex extends Exchange {
 
     async fetchBalance (params = {}) {
         await this.loadMarkets ();
+        const accountType = this.safeString (params, 'type', 'main');
+        params = this.omit (params, 'type');
+        if (accountType === 'margin') {
```

Comment:
I would suggest to split it in a bit different fashion here.

Currently, we have this, in pseudocode:

```JavaScript
fetchBalance {
    // spot code
    if (margin) {
        // margin code
        // margin code
    }
    // spot code
    // spot code
    // spot code
}
```

We could do the following:
1. rename the old fetchBalance to fetchSpotBalance and keep the body of it as it was before
2. add fetchMarginBalance alongside (lines 455-526)
3. add a new fetchBalance wrapper for both:

```JavaScript
async fetchBalance (params = {}) {
        const accountType = this.safeString (params, 'type', 'main');
        params = this.omit (params, 'type');
        if (accountType === 'margin') {
            return await this.fetchMarginBalance (params);
        } else {
            return await this.fetchSpotBalance (params);
        }
}
```